src/rgw/rgw_rados.cc

```diff
@@ -8408,8 +8408,13 @@ int RGWRados::delete_system_obj(rgw_raw_obj& obj, RGWObjVersionTracker *objv_tra
   return 0;
 }
 
-int RGWRados::delete_obj_index(const rgw_obj& obj)
+int RGWRados::delete_obj_index(const rgw_obj& _obj)
```

Comment:
But the your clearing of the instance won't take any effect outside of this function